blazingcache-core/src/main/java/blazingcache/client/CacheClient.java

```diff
@@ -1693,4 +1694,24 @@ public CacheClientStatisticsMXBean getStatistics() {
         return (T) entry.resolveReference(entrySerializer);
     }
 
+    private void locallyLockKey(RawString key) {
+        synchronized (lockedKeys) {
+            try {
+                while (!lockedKeys.add(key)) {
+                    lockedKeys.wait();
+                }
+            } catch (InterruptedException exc) {
+                Thread.currentThread().interrupt();
```

Comment:
I'd check the stopped flag and if true just exit in a clean way (the serve is stopping, we can just let the thread complete without throwing an anonymous runtime)